### PR TITLE
Refactor location select2

### PR DESCRIPTION
**Pull request by nathanathan**
This makes it so the locationSelect2 template uses the generic select2 template. It also fixes a bug where geoname lookup service urls passed in as environment vars wouldn't be used and it fixes some bugs that caused the "values" reactive vars used in select2 templates to have incorrect values.

### CODE REVIEW

File: client/controllers/incidentReports/incidentForm.coffee [hunk 1]
```diff
@@ -59,6 +60,12 @@ Template.incidentForm.onCreated ->
 
     @incidentStatus.set(@incidentData.status or '')
 
+    @locations.set(@incidentData.locations.map (loc)->
+      id: loc.id
+      text: formatLocation(loc)
+      item: loc
+    )
```
Comment (jgoley) [inline client/controllers/incidentReports/incidentForm.coffee]: Just suggesting some minor code style updates: remove the surrounding parens and add a space after the arguments?

File: client/controllers/select2.coffee [hunk 2]
```diff
@@ -45,15 +45,28 @@ Template.select2.onRendered ->
...
+    selectedValues = instance.$("select").select2('data')
+    uniqueValues = _.uniq(_.pluck(instance.values.get(), 'id'))
+    uniqueSelectedValues = _.uniq(_.pluck(selectedValues, 'id'))
+    intersection = _.intersection(uniqueValues, uniqueSelectedValues)
+    if intersection.length != uniqueSelectedValues.length or uniqueValues.length != uniqueSelectedValues.length
+      instance.values.set(selectedValues.map (data)->
+        id: data.id
+        text: data.text
+        item: data.item
+      )
```
Comment (jgoley) [inline client/controllers/select2.coffee]: Code style updates here also?

File: client/controllers/incidentReports/incidentForm.coffee (hunk 1 above)
Comment (nathanathan) [inline client/controllers/incidentReports/incidentForm.coffee]: I think the they make this a little more readable. In general, I don't think we should worry about whether parenthesis are used.

Comment (jgoley) [inline client/controllers/incidentReports/incidentForm.coffee]: 👍

**Pull request merged**